src/vsr/replica.zig

```diff
@@ -3522,11 +3522,19 @@ pub fn ReplicaType(
                         nacks.set(i);
                     }
 
+                    // Presence bit: We have the header, and the prepare is on disk,
+                    // is being written to disk, or is cached in memory.
```

Comment:
Or maybe that last line is misleading :thinking: it sort of implies that the check shouldn't consider the `journal_header`.

Reply:
:thinking: thinking even more about it, it seems like we indeed shouldn't consider `journal_header`? In the situation where:

* we have the header from persisted DVC/SV
* but we don't have journal_header because it is marked faulty
* but we none-the-less have the prepare itself (so, the slot is faulty due to redundant header being faulty)

we should signal that we are capable of reparing the prepare. 

I think it goes even further than this:

If we we have prepare matching the header in the DVC then:

* either the journal header is null
* or the journal header matches the header in the DVC

Journal header can't be newer than the header in the DVC (because DVC must included freshest info) and it can't be older than the prepare (because we update journal header before writting the prepare)